Studio-Lovelies/GG-JointJustice-Unity
# Adding a screen resolution picker

## Summary
Adds a dropdown of [all supported screen resolutions](https://docs.unity3d.com/ScriptReference/Screen-resolutions.html) to the main menu.

As with most things, this could use a visual pass, but I've built this in preparation to onboard @IsaacLaquerre, we never ended up using it and I figured discarding the code would be a waste too.

I'm happy to keep or upgrade the visuals, my suggestion would be to integrate this base implementation first and then upgrade it in another PR. Preferably that's just done by someone with a sense of design. Ergo, not me. 🙃

**Note:** Restarting the game will discard your changes; persistence is part of #231.

## Before/after screenshots and/or animated gif
<img width="549" alt="image" src="https://github.com/Studio-Lovelies/GG-JointJustice-Unity/assets/1689033/92284120-1976-43b7-88ba-35445bc876b8">

## Testing instructions
1. Start the game
2. Open the menu
3. Verify the screen resolution dropdown shows the following resolution based on your video mode:
   -  **in windowed mode:** the resolution closest* to your window size 
   -  **in fullscreen mode:** the resolution closest* to your screen resolution 
4. Select any option in the dropdown
3. Verify the selected resolution is applied to one of these based on your video mode:
   -  **in windowed mode:** your window size
   -  **in fullscreen mode:** your screen resolution

*: The game [always starts in 1280x720](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/blob/b5d039befe9b233e87e7b230c140d3bfa12f579d/unity-ggjj/ProjectSettings/ProjectSettings.asset#L47-48), even if it is not part of the [supported screen resolution list](https://docs.unity3d.com/ScriptReference/Screen-resolutions.html) for your system.

## Additional information

- `[x]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[x]` Has associated resource: 
  - #228
